src/NHibernate.Test/StaticProxyTest/StaticProxyFixture.cs

```diff
@@ -0,0 +1,477 @@
+﻿using System;
+using System.Collections;
+using System.IO;
+using System.Runtime.Serialization.Formatters.Binary;
+using NHibernate.Proxy;
+using NUnit.Framework;
+#if !NETFX
+using System.Globalization;
+using System.Runtime.Serialization;
+using NHibernate.Test.Legacy;
+using NHibernate.Util;
+#endif
+
+namespace NHibernate.Test.StaticProxyTest
+{
+	[TestFixture]
+	public class StaticProxyFixture : TestCase
+	{
+		protected override string MappingsAssembly => "NHibernate.Test";
+		protected override IList Mappings => new[] { "StaticProxyTest.Mappings.hbm.xml" };
+
+		private Guid _idSimpleEntity1;
+		private Guid _idSimpleEntity2;
+		private Guid _idInterfacedEntity1;
+		private Guid _idInterfacedEntity2;
+		private Guid _idInterfacedLazyTextEntity1;
+		private Guid _idInterfacedLazyTextEntity2;
+		private Guid _idInterfacedLazyTextEntity3;
+		private Guid _idInterfacedLazyTextEntity4;
+		private Guid _idInterfacedLazyTextEntity5;
+		private Guid _idLazyTextEntity1;
+		private Guid _idLazyTextEntity2;
+		private Guid _idLazyTextEntity3;
+		private Guid _idLazyTextEntity4;
+		private Guid _idLazyTextEntity5;
+
+		protected override void OnSetUp()
+		{
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
+			{
+				var se = new SimpleEntity
+				{
+					Name = "1",
+					Text = "Text1"
+				};
+				s.Save(se);
+				_idSimpleEntity1 = se.Id;
```

Comment:
Better to write `_idSimpleEntity1 = (Guid)s.Save(se);`. You can declare the `_idSimpleEntity1` field as object and drop casting as the concrete type of the key does not matter.

Reply:
This looks to me as quite a matter of taste. I rather have two lines of code rather than a hard-coded cast. Granted, the test case does not need the exact typing of the id.